libminifi/include/utils/StringUtils.h

```diff
@@ -330,6 +336,194 @@ class StringUtils {
     return to_hex(reinterpret_cast<const uint8_t*>(str.data()), str.length(), uppercase);
   }
 
+  /**
+   * Hexencodes a vector of bytes
+   * @param data the vector of bytes to be hexencoded
+   * @param uppercase whether the hexencoded string should be upper case
+   * @return the hexencoded string
+   */
+  inline static std::string to_hex(const std::vector<uint8_t>& data, bool uppercase = false) {
+    return to_hex(data.data(), data.size(), uppercase);
+  }
+
+  /**
+   * Decodes the Base64 encoded string into data
+   * @param data the output buffer where the decoded bytes will be written. Must be at least (base64_length / 4 + 1) * 3 bytes long.
+   * @param data_length pointer to the length of data the data buffer. It will be filled with the length of the decoded bytes.
+   * @param base64 the Base64 encoded string
+   * @param base64_length the length of base64
+   * @return true on success
+   */
+  inline static bool from_base64(uint8_t* data, size_t* data_length, const char* base64, size_t base64_length) {
```

Comment:
I think the impl of these can be moved to the correspondig cpp file.
The already existing functions were left here to reduce the impact, but I think everything newly added should have the impl in cpp file unless it's a one liner to really be inlined.

Reply:
I want to keep a few inline, but the largest ones can be moved to the cpp, you are right.